src/transforms/pcloud_func.jl

```diff
@@ -107,7 +107,7 @@ function rotate!(pcloud::PointCloud, rotmat::AbstractArray{Float32,2})
     size(rotmat) == (3, 3) || error("rotmat must be (3, 3) array, but instead got $(size(rotmat)) array")
     size(pcloud.points, 1) == 3 || error("dimension of points in PointCloud must be 3")
     points_packed = transpose(rotmat) * reshape(pcloud.points, 3, :)
-    pcloud.points = reshape(points_packed, size(pcloud.points)...)
+    pcloud.points = copy(reshape(points_packed, size(pcloud.points)...))
```

Comment:
@DhairyaLGandhi @avik-pal when using CuArray, `reshape` changes parent field of CuArray, which causes error in deepcopy. Here using `copy` degrades the performance.

Reply:
Are you thinking of a `ReshapedArray` here? That's when the parent would be transformed.

Reply:
Currently Yes. The exact error is, after reshaping, pcloud.points is `CuArray{Float32, 2, CuArray{Float32, 2, nothing}}` and when applying deepcopy on this `pcloud`, new points changes to `CuArray{Float32, 2, nothing}` and ultimately fails at deepcopy_internal as both types don't match exactly.